server/controllers/userController.js

```diff
@@ -1,107 +1,117 @@
+  createNewUser(request, response) {
+    const firstName = request.body.firstName;
+    const lastName = request.body.lastName;
+    const email = request.body.email;
+    const userName = request.body.userName;
+    const password = request.body.password;
+    if (!firstName || !lastName || !email || !userName || !password) {
+      return response.status(400).send({
+        message: 'The paramaters are incomplete',
+      });
+    }
+    db.users.findOne({ where: { $or: { email, userName } } })
       .then((userExists) => {
         if (userExists) {
-          return res.status(400).send({
-            message: 'There is a user already existing with this email'
+          return response.status(400).send({
+            message: `There is a user already existing
+            with this email or userName`
           });
         }
-        db.users
-          .create(req.body)
+        return db.users
+          .create(request.body)
           .then((user) => {
             const payload = {
               userId: user.id,
               roleId: user.roleId
             };
             const token = jwt.sign(payload, secret, { expiresIn: expires });
-            return res.status(201).send({
+            return response.status(201).send({
               message: 'User was successfully created',
               token,
               data: user
             });
           })
           .catch((err) => {
-            res.status(400).send({
-              message: err.message
+            response.status(400).send({
+              message: 'There was a problem creating this user', err
             });
           });
       });
   },
 
-  loginUser(req, res) {
+  loginUser(request, response) {
     db.users
       .findOne({
         where: {
-          email: req.body.email
+          email: request.body.email
         }
       })
       .then((user) => {
-        if (user && user.validatePassword(req.body.password)) {
+        if (!user) {
+          return response.status(400).send({
+            message: 'User was not found'
+          });
+        }
+        if (!user.validatePassword(request.body.password)) {
+          return response.status(401).send({
+            message: 'Invalid password',
+          });
+        }
+        if (user && user.validatePassword(request.body.password)) {
           const payload = {
             userId: user.id,
             roleId: user.roleId
           };
           const token = jwt.sign(payload, secret, { expiresIn: expires });
-          return res.status(200).send({
-            message: 'You were successfully logged in',
+          response.status(200).send({
+            message: 'You are successfully logged in',
             token,
             expiresIn: expires
           });
+        } else {
+          response.status(401).send({
+            message: 'Invalid login credentials',
+          });
         }
       })
       .catch((err) => {
```

Comment:
When  you're not certain about the error that will be throw, you use 500 as the status code.